src/panel/pages/explorer/components/Value.tsx

```diff
@@ -0,0 +1,127 @@
+import React, { useEffect, useRef, useMemo, FC } from "react";
+import styled from "styled-components";
+import { animated } from "react-spring";
+import { FieldNode } from "../../../context/Explorer/ast";
+import { useFlash } from "../hooks";
+
+export const Value: FC<{
+  value: FieldNode["value"];
+  expand?: boolean;
+  isRoot?: boolean;
+}> = ({ value, expand = false, isRoot = true }) => {
+  const previousValue = useRef(value);
+  const [flashProps, flash] = useFlash();
+
+  // Flash on value change
+  useEffect(() => {
+    if (isRoot && value !== previousValue.current) {
+      flash();
+    }
+
+    previousValue.current = value;
+  }, [value, flash, isRoot]);
+
+  const content = useMemo(() => {
+    const props = { value, expand };
+
+    if (value instanceof Array) {
+      return <ArrayValue {...props} value={value as any[]} />;
```

Comment:
I've broken these conditions out and we have each different state as a fixture now